test/acceptance/authz/search_test.go

```diff
@@ -39,37 +39,82 @@ func TestAuthzSearchWithGRPC(t *testing.T) {
 
 	grpcClient := helper.ClientGRPC(t)
 
-	t.Run("setup", func(t *testing.T) {
-		helper.CreateClassAuth(t, articles.ParagraphsClass(), adminKey)
-		helper.CreateClassAuth(t, articles.ArticlesClass(), adminKey)
-		helper.CreateObjectsBatchAuth(t, []*models.Object{articles.NewArticle().WithTitle("How to git gud").Object()}, adminKey)
-	})
+	tests := []struct {
+		name             string
+		mtEnabled        bool
+		tenantName       string
+		tenantPermission string
+	}{
+		{
+			name:             "with multi-tenancy",
+			mtEnabled:        true,
+			tenantName:       "tenant1",
+			tenantPermission: "tenant1",
+		},
+		{
+			name:             "without multi-tenancy",
+			mtEnabled:        false,
+			tenantName:       "",
+			tenantPermission: "*",
+		},
+	}
 
-	t.Run("correctly fail to perform a gRPC Search call without permissions", func(t *testing.T) {
-		ctx := metadata.AppendToOutgoingContext(context.Background(), "authorization", fmt.Sprintf("Bearer %s", customKey))
-		_, err := grpcClient.Search(ctx, &protocol.SearchRequest{
-			Collection: articles.ArticlesClass().Class,
-		})
-		require.NotNil(t, err)
-		require.Equal(t, status.Code(err), codes.PermissionDenied)
-	})
+	helper.CreateClassAuth(t, articles.ParagraphsClass(), adminKey)
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			defer helper.DeleteClassAuth(t, articles.ArticlesClass().Class, adminKey)
+			defer helper.DeleteClassAuth(t, articles.ArticlesClass().Class, adminKey)
+			t.Run("setup", func(t *testing.T) {
```

Comment:
you want to take this out of t.Run? Then you can run the other tests independently